src/utils/parse.js

```diff
@@ -20,51 +20,84 @@ export const filterIterations = (results, selectedParams) => {
 
 export const generateClusters = results => {
   const iterations = {};
-  const clusterGraphKeys = new Set();
   const params = new Set();
+  const runNames = new Set();
 
   Object.entries(results).forEach(([runId, result]) => {
     Object.entries(result.iterations).forEach(([, iteration]) => {
+      runNames.add(result.run_name);
       Object.entries(iteration.samples).forEach(([sampleId, sample]) => {
+        if (
+          sample.sample.measurement_title !== sample.benchmark.primary_metric ||
```

Comment:
What happens if no `measurement_title` matches the `primary_metric`?  Perhaps our indexing code should check for this condition, and the `closes_sample` condition below as well.